hs-bindgen/src-internal/HsBindgen/Config/Internal.hs

```diff
@@ -87,6 +91,8 @@ data FrontendConfig = FrontendConfig {
 data BackendConfig = BackendConfig {
       backendTranslationConfig :: TranslationConfig
     , backendHaddockConfig     :: HaddockConfig
+    , backendOutputDirPolicy   :: OutputDirPolicy
```

Comment:
I think it is OK to have the policies in the backend configuration. On the other hand, then they have to be provided even when we do not create output files (e.g., TH mode).

We could also provide them with the respective artefacts, but then we have to do so for every artefact involving file output. What do you think?